### PR TITLE
feat: polyfill `node:worker_threads` module

### PR DESCRIPTION
**Pull request by jculvey**
Replaces the current auto-mocking of `worker_threads` to support destructured ESM imports and allow for functional polyfill coverage in the future.

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by IgorMinar — APPROVED** [review]
this looks very reasonable to me. thanks James!

File: src/runtime/node/worker_threads/index.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import { Serializable } from "node:worker_threads";
+
+export { BroadcastChannel } from "./broadcast-channel";
+export { MessageChannel } from "./message-channel";
+export { MessagePort } from "./message-port";
+export { Worker } from "./worker";
+
+export const getEnvironmentData: typeof worker_threads.getEnvironmentData =
+  () => {
+    return undefined as unknown as Serializable;
```
Comment (pi0) [inline src/runtime/node/worker_threads/index.ts:17]: Can get/set could be coupled into working state?

File: src/runtime/node/worker_threads/index.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  };
+export const isMainThread: typeof worker_threads.isMainThread = false;
```
Comment (pi0) [inline src/runtime/node/worker_threads/index.ts:19]:
```suggestion
export const isMainThread: typeof worker_threads.isMainThread = true;
```

**Review by pi0 — APPROVED** [review]
Thanks! (pushed some last updates)

**Pull request merged**